File: apps/spring-boot-api/src/main/java/org/bootstrapbugz/api/shared/error/exception/ForbiddenException.java
package org.bootstrapbugz.api.shared.error.exception;

import lombok.Getter;
import org.springframework.http.HttpStatus;

import java.io.Serial;

@Getter
public class ForbiddenException extends RuntimeException {
  @Serial private static final long serialVersionUID = 6438756026918720197L;
  private final HttpStatus status;

  public ForbiddenException(String message) {
    super(message);
    this.status = HttpStatus.FORBIDDEN;
  }
}

File: backend/spring-boot-kotlin/src/main/java/org/bootstrapbugz/api/shared/error/exception/ForbiddenException.kt
package org.bootstrapbugz.api.shared.error.exception

import org.springframework.http.HttpStatus
import java.io.Serial

class ForbiddenException(message: String?) : RuntimeException(message) {
    val status: HttpStatus = HttpStatus.FORBIDDEN

    companion object {
        @Serial
        private val serialVersionUID = 6438756026918720197L
    }
}
